engine/table/src/main/java/io/deephaven/engine/table/impl/updateby/rollingstd/CharRollingStdOperator.java

```diff
@@ -0,0 +1,146 @@
+                        return a + b;
+                    }));
+            valueSquareBuffer = new AggregatingDoubleRingBuffer(BUFFER_INITIAL_CAPACITY, 0.0,
+                    Double::sum,
+                    ((a, b) -> {
+                        if (a == NULL_DOUBLE && b == NULL_DOUBLE) {
+                            return 0.0; // identity value
+                        } else if (a == NULL_DOUBLE) {
+                            return b;
+                        } else if (b == NULL_DOUBLE) {
+                            return a;
+                        }
+                        return a + b;
+                    }));
+        }
+
+        @Override
+        public void close() {
+            super.close();
+            valueBuffer = null;
+        }
+
+        @Override
+        public void setValuesChunk(@NotNull final Chunk<? extends Values> valuesChunk) {
+            influencerValuesChunk = valuesChunk.asCharChunk();
+        }
+
+        @Override
+        public void push(int pos, int count) {
+            valueBuffer.ensureRemaining(count);
+            valueSquareBuffer.ensureRemaining(count);
+
+            for (int ii = 0; ii < count; ii++) {
+                final char val = influencerValuesChunk.get(pos + ii);
+
+                if (val != NULL_CHAR) {
+                    // Add the value and its square to the buffers.
+                    valueBuffer.addUnsafe(val);
+                    valueSquareBuffer.addUnsafe((double)val * val);
+                } else {
+                    // Add null to the buffers and increment the count.
+                    valueBuffer.addUnsafe(NULL_DOUBLE);
+                    valueSquareBuffer.addUnsafe(NULL_DOUBLE);
+                    nullCount++;
+                }
+            }
+        }
+
+        @Override
+        public void pop(int count) {
+            Assert.geq(valueBuffer.size(), "valueBuffer.size()", count);
+            Assert.geq(valueSquareBuffer.size(), "valueSquareBuffer.size()", count);
+
+            for (int ii = 0; ii < count; ii++) {
+                final double val = valueBuffer.removeUnsafe();
+                final double valSquare = valueSquareBuffer.removeUnsafe();
+
+                if (val == NULL_DOUBLE || valSquare == NULL_DOUBLE) {
+                    nullCount--;
+                }
+            }
+        }
+
+        @Override
+        public void writeToOutputChunk(int outIdx) {
+            if (valueBuffer.size() == 0) {
+                outputValues.set(outIdx, NULL_DOUBLE);
+            } else {
+                final int count = valueBuffer.size() - nullCount;
+                final double valueSquareSum = valueSquareBuffer.evaluate();
+                final double valueSum = valueBuffer.evaluate();
+
+                final double variance = valueSquareSum / (count - 1) - valueSum * valueSum / count / (count - 1);
+                final double std = Math.sqrt(variance);
+
+                outputValues.set(outIdx, std);
```

Comment:
Is this correct? `count - 1` seems suspicious. A std-deviation of a single value is zero, not null (this impl will div by zero; but I see you protected against divByZero in `BigDecimal`/`BigInteger`).

I believe the variance should be: `valueSquareSum / count - (valueSum / count) ^ 2`

Reply:
This matches the implementation of `var` found in the `Numeric` classes. See `io.deephaven.function.Numeric#var(io.deephaven.vector.ByteVector)` for an example.

I was under the impression we were returning `unbiased sample variance` as explained here: https://en.wikipedia.org/wiki/Bessel%27s_correction.  @chipkent, is this correct?

Reply:
The wikipedia article is correct.  Everyone thinks it is wrong initially.